core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -105,7 +111,14 @@ final case class Uri(
 
       case Uri(None, None, _, _, _) =>
     }
-    writer << path
+
+    this match {
```

Comment:
What if there is a scheme and no authority?  I guess this behavior is scheme specific?

Reply:
I do believe you need an authority for the path segmenting to make sense.

Reply:
`file://` is the one that jumped to my mind.  But not everything has a `/`-rooted filesystem.